pkg/autoscaler/metrics/stats_scraper.go

```diff
@@ -132,6 +137,64 @@ func urlFromTarget(t, ns string) string {
 		t, ns, networking.AutoscalingQueueMetricsPort)
 }
 
+// ScrapeForRemoval collects metrics from all pods for decision making on removal
+func (s *ServiceScraper) ScrapeForRemoval(readyCount, desiredScale int) ([]string, error) {
+	removalCount := readyCount - desiredScale
+	sampleSize := readyCount
+
+	statChan := make(chan Stat, sampleSize)
+	err := s.doScrape(statChan, sampleSize)
+	if err != nil {
+		return nil, err
+	}
+
+	close(statChan)
+
+	var podInfo []Stat
+	for stat := range statChan {
+		podInfo = append(podInfo, stat)
+	}
+
+	sort.SliceStable(podInfo, func(i, j int) bool {
+		return podInfo[i].AverageConcurrentRequests+podInfo[i].AverageProxiedConcurrentRequests <
+			podInfo[j].AverageConcurrentRequests+podInfo[j].AverageProxiedConcurrentRequests
+	})
+
+	podsToRemove := make([]string, 0, removalCount)
+	for _, p := range podInfo[:removalCount] {
+		podsToRemove = append(podsToRemove, p.PodName)
+	}
+
+	return podsToRemove, nil
+}
+
+func (s *ServiceScraper) doScrape(statCh chan Stat, sampleSize int) error {
+	scrapedPods := &sync.Map{}
+	grp := errgroup.Group{}
+
+	for i := 0; i < sampleSize; i++ {
+		grp.Go(func() error {
+			for tries := 1; ; tries++ {
+				stat, err := s.tryScrape(scrapedPods)
```

Comment:
It would be beneficial to actually pick older pods than newer as we do for scaling scraping.

Reply:
here we are looping over everything. but when sorting them out, I can make it so that older pods are preferred over the new ones when scaling down.